python-sdk/nuscenes/map_expansion/arcline_path_utils.py

```diff
@@ -0,0 +1,288 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+
+import math
+from typing import Dict, Any, Tuple, List
+
+import numpy as np
+
+# (x, y, yaw) in global frame
+Pose = Tuple[float, float, float]
+
+ArcLinePath = Dict[str, Any]
+
+
+def principal_value(angle_in_radians: float) -> float:
+    """
+    Ensures the angle is within [-pi, pi).
+    :param angle_in_radians: Angle in radians.
+    :return: Scaled angle in radians.
+    """
+
+    interval_min = -math.pi
+    two_pi = 2 * math.pi
+    scaled_angle = (angle_in_radians - interval_min) % two_pi + interval_min
+    return scaled_angle
+
+
+def compute_segment_sign(arcline_path: ArcLinePath) -> Tuple[int, int, int]:
+    """
+    Compute the sign of an arcline path based on its shape.
+    :param arcline_path: arcline path record.
+    :return: Tuple of signs for all three parts of the path. 0 if straight, -1 if right,
+        1 if left.
+    """
+    shape = arcline_path['shape']
+    segment_sign = [0, 0, 0]
+
+    if shape in ("LRL", "LSL", "LSR"):
+        segment_sign[0] = 1
+    else:
+        segment_sign[0] = -1
+
+    if shape == "RLR":
+        segment_sign[1] = 1
+    elif shape == "LRL":
+        segment_sign[1] = -1
+    else:
+        segment_sign[1] = 0
+
+    if shape in ("LRL", "LSL", "RSL"):
+        segment_sign[2] = 1
+    else:
+        segment_sign[2] = -1
+
+    return segment_sign[0], segment_sign[1], segment_sign[2]
+
+
+def get_transformation_at_s(pose: Pose,
+                            s: float) -> Pose:
+    """
+    Get the affine transformation at s meters along the path.
+    :param pose: Pose represented as tuple (x, y, yaw).
+    :param s: Length along the arcline path in range (0, length_of_arcline_path].
```

Comment:
Inconsistent brackets: (]

Reply:
The right end point is supposed to be inclusive, so I used `]`